backend/endpoints/scan.py

```diff
@@ -86,7 +104,106 @@ async def scan_platforms(
+            scanned_state = scan_state(
+                platform=scanned_platform,
+                emulator=fs_emulator,
+                file_name=fs_state_filename,
+            )
+
+            state = dbh.get_state_by_filename(scanned_platform.slug, fs_state_filename)
+            if state:
+                dbh.update_state(
+                    state.id, {"file_size_bytes": scanned_state.file_size_bytes}
+                )
+                continue
+
+            scanned_state.emulator = fs_emulator
+            scanned_state.platform_slug = scanned_platform.slug
+
+            rom = dbh.get_rom_by_filename_no_tags(scanned_state.file_name_no_tags)
+            if rom:
+                scanned_state.rom_id = rom.id
+                dbh.add_state(scanned_state)
+
+        # Scanning screenshots
+        log.info(f"\t · {len(fs_assets['screenshots'])} screenshots found")
+        for fs_screenshot_filename in fs_assets["screenshots"]:
+            scanned_screenshot = scan_screenshot(
+                file_name=fs_screenshot_filename, fs_platform=scanned_platform.slug
+            )
+
+            screenshot = dbh.get_screenshot_by_filename(fs_screenshot_filename)
+            if screenshot:
+                dbh.update_screenshot(
+                    screenshot.id,
+                    {"file_size_bytes": scanned_screenshot.file_size_bytes},
+                )
+                continue
+
+            scanned_screenshot.platform_slug = scanned_platform.slug
+
+            rom = dbh.get_rom_by_filename_no_tags(scanned_screenshot.file_name_no_tags)
+            if rom:
+                scanned_screenshot.rom_id = rom.id
+                dbh.add_screenshot(scanned_screenshot)
+
+        dbh.purge_saves(scanned_platform.slug, [s for _e, s in fs_assets["saves"]])
+        dbh.purge_states(scanned_platform.slug, [s for _e, s in fs_assets["states"]])
+        dbh.purge_screenshots(fs_assets["screenshots"], scanned_platform.slug)
         dbh.purge_roms(scanned_platform.slug, [rom["file_name"] for rom in fs_roms])
+
+    # Scanning screenshots outside platform folders
+    fs_screenshots = get_screenshots()
+    log.info("Screenshots")
+    log.info(f" · {len(fs_screenshots)} screenshots found")
+    for fs_platform, fs_screenshot_filename in fs_screenshots:
+        scanned_screenshot = scan_screenshot(
+            file_name=fs_screenshot_filename, fs_platform=fs_platform
+        )
+
+        screenshot = dbh.get_screenshot_by_filename(fs_screenshot_filename)
+        if screenshot:
+            dbh.update_screenshot(
+                screenshot.id, {"file_size_bytes": scanned_screenshot.file_size_bytes}
+            )
+            continue
+
+        rom = dbh.get_rom_by_filename_no_tags(scanned_screenshot.file_name_no_tags)
+        if rom:
+            scanned_screenshot.rom_id = rom.id
+            scanned_screenshot.platform_slug = rom.platform_slug
+            dbh.add_screenshot(scanned_screenshot)
+
```

Comment:
This is for screenshots manually added, or where the user might not place screenshots under a specific platform slug